<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv='Content-Type' content='text/html; charset=utf-8' />
    <title>James Hensman</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="/js/jquery.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <link href="/css/bootstrap.min.css" rel="stylesheet">
    <link href="/css/theme.css" rel="stylesheet">
</head>

<body>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                  <span class="sr-only">Toggle navigation</span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="/">James Hensman</a>
              </div>
              <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li class="active"><a href="/">Home</a></li>
                    <li class="active visible-xs-block"><a href="/links.html">Links</a></li>
                    <li class="active"><a href="/code.html">Code</a></li>
                    <li class="active"><a href="/publications.html">Publications</a></li>
                </ul>
              </div>
        </div>
    </div>
</div>


<div class="container container-left">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class="sidebar well">
<td align="right"> <img src='images/me_cropped.jpg'></td>
</div>


<div class="sidebar well">
	<h1>Links</h1>
  <li><a href="http://twitter.com/jameshensman">twitter</a></li>
  <li><a href="http://github.com/jameshensman">github</a></li>
  <li><a href="https://scholar.google.co.uk/citations?user=l8dX3ssAAAAJ&hl=en">scholar</a></li>
</div>

        </div>
        <div class="col-md-9">
		<div class="article">
		    <div class="well">
			<div class="content">
			    <h1 id="gpflow">GPflow</h1>
<p>With <a href="http://mlg.eng.cam.ac.uk/?portfolio=alex-matthews">Alex Matthews</a> and many others I develop and maintain <a href="http://github.com/GPflow/GPflow">GPflow</a>, A Gaussian process library based on <a href="http://tensorflow.org">TensorFlow</a>.</p>

<h1 id="gpy">GPy</h1>
<p>In 2010 I started working on GPy. Based in part on Neil Lawrence’s matlab code, I started to make a framework for Gaussian processes in python. The aim was to facilitate reproducible research, and to give myself a platform to build future work on, with things like automatic gradient-checking built in.</p>

<p>Now GPy has evolved into much more than that. With the contribution of other lab members, we began to build a platform that can be used for research and teaching.</p>

<p>Check out the code on <a href="http://github.com/sheffieldml/gpy">github</a></p>

<h1 id="gpclust">GPclust</h1>
<p>GPclust contains effiecient code for clustering time series using Gaussian processes. It’s based on a modification to the VBEM algorithm that I wrote about <a href="http://papers.nips.cc/paper/4766-fast-variational-inference-in-the-conjugate-exponential-family">here</a>, and builds on GPy. There are <a href="https://github.com/jameshensman/GPclust/blob/master/notebooks/index.ipynb">notebook tutorials</a> to go with the <a href="https://github.com/jameshensman/GPclust/">open source code</a></p>

<h1 id="hierarchical-gps">Hierarchical GPs</h1>
<p>It’s possible to extend Gaussian processes into hierarchical structures to model groupings in data. The linear-Gaussian mature of the model means that the whole thing is still a Gaussian process: I wrote about this in <a href="http://www.biomedcentral.com/1471-2105/14/252">BMC Bioinformatics</a>. <br />
I’ve incorporated the code for this paper into GPy. I’ve created an IPython notebook to show how to <a href="https://github.com/SheffieldML/notebook/blob/master/compbio/Hierarchical.ipynb">apply these ideas</a></p>

<h1 id="variational-inference">Variational Inference</h1>
<p>In 2012 I wrote about a method for doing <a href="">fast optimization in Variational Bayes problems</a>. It’s a generally applicable method, so I’ve used the code for other models also. You can find the code on <a href="githib.com/sheffieldml/colvb">github</a>.</p>

<h1 id="rna-transcript-quantification-bitseq">RNA Transcript Quantification (BitSeq)</h1>
<p>Variational inference is useful across so many domains: we’ve recently shown how it can be aplied to count molucues from high-throuput sequencing. The paper is <a href="http://bioinformatics.oxfordjournals.org/content/early/2015/09/04/bioinformatics.btv483.full.pdf+html">here</a> and the code is now part of the BitSeq project, <a href="https://www.bioconductor.org/packages/release/bioc/html/BitSeq.html">here</a>.</p>


			</div>
		    </div>
		</div>
        </div>
    </div>
</div>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2017 James Hensman. Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>






</body>
</html>

